modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayLink from '@clayui/link';
+import ClaySticker from '@clayui/sticker';
+import ClayTabs from '@clayui/tabs';
+import React, {useMemo, useState} from 'react';
+
+import Sidebar from './Sidebar';
+
+const formatDate = (date, languageTag) => {
+	return (
+		date &&
+		languageTag &&
+		Intl.DateTimeFormat(languageTag, {
+			day: 'numeric',
+			hour: 'numeric',
+			hour12: true,
+			minute: 'numeric',
+			month: 'short',
+			year: 'numeric',
+		}).format(new Date(date))
+	);
+};
+
+const SidebarPanelInfoView = ({
+	categories = [],
+	classPK,
+	createDate,
+	data = {},
+	languageTag = 'en',
+	modifiedDate,
+	subType,
+	tags = [],
+	title,
+	userName,
+	userPortraitURL,
+	versions = [],
+	viewURLs = [],
+}) => {
+	const [activeTabKeyValue, setActiveTabKeyValue] = useState(0);
+
+	const formattedCreateDate = useMemo(() => {
+		return formatDate(createDate, languageTag);
+	}, [createDate, languageTag]);
+
+	const formattedDisplayDate = useMemo(() => {
+		return formatDate(data['display-date']?.value, languageTag);
+	}, [data, languageTag]);
+
+	const formattedExpirationDate = useMemo(() => {
+		return formatDate(data['expiration-date']?.value, languageTag);
+	}, [data, languageTag]);
+
+	const formattedModifiedDate = useMemo(() => {
+		return formatDate(modifiedDate, languageTag);
+	}, [modifiedDate, languageTag]);
+
+	const formattedReviewDate = useMemo(() => {
+		return formatDate(data['review-date']?.value, languageTag);
+	}, [data, languageTag]);
```

Comment:
To reduce some boilerplate, could we use implicit returns here?

I remember there being a discussion about implicit returns and SF, but I don't fully remember and didn't see it in the frontend-guidelines repo. So maybe we enforce not using implicit returns.

Reply:
Also, I wonder if we need all these functions or if we could get away with just directly calling `formatDate()` inline.

Reply:
Don't know if there will be SF problems, will try

Reply:
> Also, I wonder if we need all these functions or if we could get away with just directly calling formatDate() inline.

I added just to not format all dates on every render. We can get rid of them if we consider the performance gain doesn't worth it. What do you think guys?

Reply:
We could still use `useMemo`, but on a single function like:

```js
const getFormattedDate = useMemo(() => formatDate(value, languageTag), [value, languageTag]);
```

But yeah, the general rule of thumb I hear people recommending `useMemo` for is for expensive operations like iterating over an array or something like it. Not sure it applies here.

Reply:
I'm ok with do it inline, I'll add a commit for it, if somebody have another / stronger opinion I can just drop it 😄

Reply:
Some tips on when to use `useMemo`:

- If in doubt, profile: countless bothans have died optimizing code that wasn't a performance bottleneck (or not optimizing code that _was_ a bottleneck) due to not actually measuring where time is being spent.
- Having said that, if you want to make a ballpark estimation of how expensive an operation is relative to memoizing it, you need to understand what the operation is doing _and_ also have an idea of what `useMemo` would be doing instead; the first one varies by case, but we can say something general about the second one:
  - `useMemo` has some time overhead because it is doing an `Object.is()` check on each item in the dependency array, and some space overhead because it is storing "metadata" which includes a copy of the callback function, and is likely doing so by using a `useRef()` under the covers. Additionally, there is some short-lived garbage created because you create an arrow function and a dependencies array on every render, and a miniscule function call overhead from calling `useMemo()` itself. Basically, if the thing you're doing is more expensive than all of that, `useMemo()` will be a win.